scheduler/src/cook/mesos/data_locality.clj

```diff
@@ -0,0 +1,30 @@
+(ns cook.mesos.data-locality
+  (:require [cook.config :as config])
+  (:import com.netflix.fenzo.VMTaskFitnessCalculator
+           com.netflix.fenzo.plugins.BinPackingFitnessCalculators))
+
+(def data-local-costs (atom {}))
+
+(defn get-normalized-fitness
+  "Returns the fitness for the job to run on the given host, normalized by max-cost"
+  [uuid hostname max-cost]
+  (let [cost (get-in @data-local-costs [uuid hostname] max-cost)]
+    (- 1 (double (/ cost max-cost)))))
+
+(deftype DataLocalFitnessCalculator [^VMTaskFitnessCalculator base-calculator data-locality-weight maximum-cost]
```

Comment:
I would prefer `DataLocalFitnessCalculator` returning only the data locality cost directly and not having a reference to `base-calculator`.  We can have a separate `WeightedAverageFitnessCalculator` that contains the reference to both `base-calculator` and `data-local-calculator` to compute the weighted average fitness.